src/schema/posts.ts

```diff
@@ -833,6 +854,48 @@ export const resolvers: IResolvers<any, Context> = {
       });
       return { _: true };
     },
+    submitExternalLink: async (
+      _,
+      {
+        sourceId,
+        url,
+        commentary,
+      }: { sourceId: string; url: string; commentary: string },
+      ctx,
+    ): Promise<GQLEmptyResponse> => {
+      await ensureSourcePermissions(ctx, sourceId, SourcePermissions.Post);
+      const cleanUrl = standardizeURL(url);
+      if (!isValidHttpUrl(cleanUrl)) {
+        throw new ValidationError('URL is not valid');
+      }
+
+      const existingPost = await ctx.con.getRepository(ArticlePost).findOne({
+        select: ['id'],
+        where: [
+          { url: cleanUrl, deleted: false },
+          { canonicalUrl: cleanUrl, deleted: false },
+        ],
+      });
+      if (existingPost) {
+        await createSharePost(
+          ctx.con,
+          sourceId,
+          ctx.userId,
+          existingPost.id,
+          commentary,
+        );
+        return { _: true };
+      }
+      await createExternalLink(
+        ctx.con,
+        ctx.log,
+        sourceId,
+        ctx.userId,
+        url,
+        commentary,
+      );
```

Comment:
This needs to be in a transaction

Reply:
@idoshamun Converted it to be wrapped and have the main functions accept the manager.